common/web/gesture-recognizer/src/engine/headless/gestureDebugPath.ts

```diff
@@ -0,0 +1,140 @@
+import { InputSample } from "./inputSample.js";
+import { CumulativePathStats } from "./cumulativePathStats.js";
+import { Mutable } from "../mutable.js";
+import { GesturePath } from "./gesturePath.js";
+
+/**
+ * Documents the expected typing of serialized versions of the `GesturePath` class.
+ */
+export type SerializedGesturePath<Type, StateToken> = {
+  coords: Mutable<InputSample<Type, StateToken>>[]; // ensures type match with public class property.
+  wasCancelled?: boolean;
+}
+
+interface EventMap<Type, StateToken> {
+  'step': (sample: InputSample<Type, StateToken>) => void,
+  'complete': () => void,
+  'invalidated': () => void
+}
+
+/**
+ * Models the path over time through coordinate space taken by a touchpoint during
+ * its active lifetime.
+ *
+ *  _Supported events_:
```

Comment:
Right, this started off by duplicating `GesturePath`.  

Note that since `GesturePath` is the parent class, this class actually does provide those events as well.